presentation/common/src/presentation-common/RpcRequestsHandler.ts

```diff
@@ -199,4 +202,29 @@ export class RpcRequestsHandler implements IDisposable {
     return this.request<KeySetJSON, SelectionScopeRequestOptions<IModelRpcProps>>(
       this.rpcClient.computeSelection.bind(this.rpcClient), options, ids, scopeId);
   }
+
+  private static removeUnneededPropertiesFromRuleset(ruleset: any) {
+    type RulesetWithRequiredProperties = {
+      [key in keyof Ruleset]-?: true;
+    };
+
+    const rulesetWithRequiredProperties: RulesetWithRequiredProperties = {
+      id: true,
+      rules: true,
+      version: true,
+      requiredSchemas: true,
+      supplementationInfo: true,
+      vars: true,
+    };
+
+    for (const propertyKey of Object.keys(ruleset)) {
+      if (!rulesetWithRequiredProperties.hasOwnProperty(propertyKey)) {
+        if (propertyKey === "$schema")
+          delete ruleset[propertyKey];
```

Comment:
Caller may have certain expectations what the ruleset object contains, thus it's unwise to mutate the passed-in object. It's better to create a new one and copy the top-level properties over.

Reply:
good point, let's create a new one instead of mutating the original